src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,110 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private K key;
+    private int size;
+    private int threshold;
+
+    private static class Node<K,V> {
+        private final K key;
+        private V value;
+        private Node<K,V> next;
+
+        public Node(K key, V value, Node<K,V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        threshold = (int) (DEFAULT_LOAD_FACTOR * DEFAULT_INITIAL_CAPACITY);
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K,V> node = new Node<>(key, value, null);
+        if (table[getIndex()] == null) {
+            table[getIndex()] = node;
+        } else {
+            Node<K,V> current = table[getIndex()];
+            while (current != null) {
+                if (Objects.equals(current.key,node.key)) {
+                    current.value = node.value;
+                    return;
+                }
+                if (current.next == null) {
+                    current.next = node;
+                    break;
+                }
+                current = current.next;
+            }
+        }
+        if (size++ > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        V value = null;
+        Node<K,V> current = table[getIndex()];
+        while (current != null) {
+            if (Objects.equals(current.key,key)) {
+                value = current.value;
+            }
+            current = current.next;
+        }
+        return value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public int getIndex() {
+        return hash(this.key) % table.length;
+    }
```

Comment:
```suggestion
    public int getTableIndex(K key) {
        return hash(key) % table.length;
    }
```

Принцип работы HashMap следующий:
	- put(K key, V value) -> Мы ищем по хешкоду ключа в какой бакет(индекс массива table) должна попасть нода, далее проверяем не лежит ли в этой ячейке массива table уже какая-то нода, если нет, то кладем новую ноду, если есть то проверяем нету ли такого же ключа в связаном списке нод, который лежит в этой ячейке, если нету - пристегиваем новую ноду в конец, если есть - перезатираем значение
	- get(K key) -> Мы ищем по хешкоду ключа в какой бакет(ячейка массива table с индексом, который расчитывается по хешкоду ключа) должна была попасть нода с таким ключем, далее смотрим есть ли в этом бакете односвязный список нод, если есть -> проходимся по нему и ищем ноду с таким же ключем, как приходит нам на вход. Если не нашли такую ноду -> возвращаем null

Для этих методов, нам нужен метод `int getTableIndex(K key)` ***который будет принимать на вход ключ*** , и по его хешкоду определять в какой бакет попадет нода с таким ключем.

Reply:
Дарія, дуже дякую що звернула увагу, вже вникаю і розбираюсь, просто робота трохи прижала то динаміка стопорнулась.